coderd/prometheusmetrics/aggregator.go

```diff
@@ -118,10 +139,48 @@ func (am *annotatedMetric) asPrometheus() (prometheus.Metric, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	return prometheus.MustNewConstMetric(desc, valueType, am.Value, labelValues...), nil
 }
 
-func NewMetricsAggregator(logger slog.Logger, registerer prometheus.Registerer, duration time.Duration) (*MetricsAggregator, error) {
+// getFieldByLabel returns the related field value for a given label
+func (am *annotatedMetric) getFieldByLabel(label string) (string, error) {
+	var labelVal string
+	switch label {
+	case agentmetrics.WorkspaceNameLabel:
+		labelVal = am.workspaceName
+	case agentmetrics.TemplateNameLabel:
+		labelVal = am.templateName
+	case agentmetrics.AgentNameLabel:
+		labelVal = am.agentName
+	case agentmetrics.UsernameLabel:
+		labelVal = am.username
+	default:
+		return "", xerrors.Errorf("unexpected label: %q", label)
+	}
+
+	return labelVal, nil
+}
+
+func (am *annotatedMetric) clone() annotatedMetric {
```

Comment:
Would `shallowCopy` be more appropriate as a name, or maybe `cloneShallow`? I'm thinking it would be clearer since we're not cloning slices. Alt: should this be a deep clone?